#  WARNING: DO NOT RUN THIS FILE !
#  NOTE: This Jupyter Notebook is not supposed to be RUN. It is created only for Documentation purposes.

# 1. Objective

To develop an analytical dashboard using Streamlit that explores Adidas sales data, providing visualizations and insights to identify patterns, offer insightful commentary, and highlight areas for potential improvement. The app will present a thorough study of Adidas' sales performance over a specified period, enabling users to interact with the data and gain valuable business insights.

# 1.1. Data Description

* Retailer: The name of the retailer who sold the Adidas product
* Retailer ID: The unique identifier for the retailer
* Invoice Date: The date the invoice was issued for the sale of the Adidas product
* Region: The geographic region where the retailer is located
* State: The state where the retailer is located
* City: The city where the retailer is located
* Product: The name of the Adidas product that was sold
* Price per Unit: The price of the Adidas product per unit
* Units Sold: The number of units of the Adidas product that were sold
* Total Sales: The total sales for the Adidas product, calculated as the price per unit multiplied by the units sold
* Operating Profit: The operating profit for the sale of the Adidas product
* Operating Margin: The operating margin for the sale of the Adidas product, calculated as the operating profit divided by the total sales
* Sales Method: The method by which the Adidas product was sold, such as online or in-store

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import warnings
warnings.filterwarnings('ignore')
import plotly.graph_objects as go

In [2]:
df=pd.read_excel('Adidas US Sales Datasets.xlsx')
df

Unnamed: 0   Unnamed: 1             Unnamed: 2           Unnamed: 3  \
0            NaN          NaN  Adidas Sales Database                  NaN   
1            NaN          NaN                    NaN                  NaN   
2            NaN          NaN                    NaN                  NaN   
3            NaN     Retailer            Retailer ID         Invoice Date   
4            NaN  Foot Locker                1185732  2020-01-01 00:00:00   
...          ...          ...                    ...                  ...   
9647         NaN  Foot Locker                1185732  2021-01-24 00:00:00   
9648         NaN  Foot Locker                1185732  2021-01-24 00:00:00   
9649         NaN  Foot Locker                1185732  2021-02-22 00:00:00   
9650         NaN  Foot Locker                1185732  2021-02-22 00:00:00   
9651         NaN  Foot Locker                1185732  2021-02-22 00:00:00   

     Unnamed: 4     Unnamed: 5  Unnamed: 6               Unnamed: 7  \
0           NaN            NaN         NaN                      NaN   
1           NaN            NaN         NaN                      NaN   
2           NaN            NaN         NaN                      NaN   
3        Region          State        City                  Product   
4     Northeast       New York    New York    Men's Street Footwear   
...         ...            ...         ...                      ...   
9647  Northeast  New Hampshire  Manchester            Men's Apparel   
9648  Northeast  New Hampshire  Manchester          Women's Apparel   
9649  Northeast  New Hampshire  Manchester    Men's Street Footwear   
9650  Northeast  New Hampshire  Manchester  Men's Athletic Footwear   
9651  Northeast  New Hampshire  Manchester  Women's Street Footwear   

          Unnamed: 8  Unnamed: 9  Unnamed: 10       Unnamed: 11  \
0                NaN         NaN          NaN               NaN   
1                NaN         NaN          NaN               NaN   
2                NaN         NaN          NaN               NaN   
3     Price per Unit  Units Sold  Total Sales  Operating Profit   
4                 50        1200       600000            300000   
...              ...         ...          ...               ...   
9647              50          64         3200             896.0   
9648              41         105         4305            1377.6   
9649              41         184         7544           2791.28   
9650              42          70         2940            1234.8   
9651              29          83         2407            649.89   

           Unnamed: 12   Unnamed: 13  
0                  NaN           NaN  
1                  NaN           NaN  
2                  NaN           NaN  
3     Operating Margin  Sales Method  
4                  0.5      In-store  
...                ...           ...  
9647              0.28        Outlet  
9648              0.32        Outlet  
9649              0.37        Outlet  
9650              0.42        Outlet  
9651              0.27        Outlet  

[9652 rows x 14 columns]

# 1.2. Data Cleaning

* Removing Unnecessary Rows and Columns that contained irrelevant information
* Setting Column Names
* The index was reset to ensure a continuous sequence of numbers for better organization
* Checking for duplicate rows within the dataset to identify and potentially remove redundant data
* Converting Data Types to the appropriate format for accurate date-based analysis
* Extracting Date Components to facilitate date-based filtering and analysis

In [3]:
#Removing First 3 rows
df = df.drop(df.index[0:3]) 

#Dropping first columns
df.drop("Unnamed: 0", axis = 1, inplace = True) 

#Changing the column names
df.columns = df.iloc[0] 

#Dropping the first row 
df = df.drop(df.index[0]) 

#Reseting the index 
df = df.reset_index(drop=True) 

#Checking whether the dataset have duplicated values
df.duplicated().sum() 

0

In [4]:
df.describe()

3          Retailer  Retailer ID         Invoice Date Region       State  \
count          9648         9648                 9648   9648        9648   
unique            6            4                  724      5          50   
top     Foot Locker      1185732  2021-01-17 00:00:00   West  California   
freq           2637         5265                   77   2448         432   

3           City                Product  Price per Unit  Units Sold  \
count       9648                   9648            9648        9648   
unique        52                      6             146         361   
top     Portland  Men's Street Footwear              50         225   
freq         360                   1610             557         207   

3       Total Sales  Operating Profit  Operating Margin Sales Method  
count          9648              9648           9648.00         9648  
unique         3512              6510            127.00            3  
top          100000             21000              0.35       Online  
freq             52                34           1229.00         4889

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9648 entries, 0 to 9647
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Retailer          9648 non-null   object
 1   Retailer ID       9648 non-null   object
 2   Invoice Date      9648 non-null   object
 3   Region            9648 non-null   object
 4   State             9648 non-null   object
 5   City              9648 non-null   object
 6   Product           9648 non-null   object
 7   Price per Unit    9648 non-null   object
 8   Units Sold        9648 non-null   object
 9   Total Sales       9648 non-null   object
 10  Operating Profit  9648 non-null   object
 11  Operating Margin  9648 non-null   object
 12  Sales Method      9648 non-null   object
dtypes: object(13)
memory usage: 980.0+ KB


In [6]:
#Changing datatype of Invoice Date to datetime
df['Invoice Date']=pd.to_datetime(df['Invoice Date'])

#Changing the datatype of Price per Unit', 'Units Sold', 'Total Sales','Operating Profit', 'Operating Margin' columns to float
df[['Price per Unit', 'Units Sold', 'Total Sales','Operating Profit', 'Operating Margin']] = df[['Price per Unit', 'Units Sold', 'Total Sales','Operating Profit', 'Operating Margin']].astype("float")
df['Year'] = df['Invoice Date'].dt.year
df['Month'] = df['Invoice Date'].dt.month
df['Day'] = df['Invoice Date'].dt.day

df.head()

3     Retailer Retailer ID Invoice Date     Region     State      City  \
0  Foot Locker     1185732   2020-01-01  Northeast  New York  New York   
1  Foot Locker     1185732   2020-01-02  Northeast  New York  New York   
2  Foot Locker     1185732   2020-01-03  Northeast  New York  New York   
3  Foot Locker     1185732   2020-01-04  Northeast  New York  New York   
4  Foot Locker     1185732   2020-01-05  Northeast  New York  New York   

3                    Product  Price per Unit  Units Sold  Total Sales  \
0      Men's Street Footwear            50.0      1200.0     600000.0   
1    Men's Athletic Footwear            50.0      1000.0     500000.0   
2    Women's Street Footwear            40.0      1000.0     400000.0   
3  Women's Athletic Footwear            45.0       850.0     382500.0   
4              Men's Apparel            60.0       900.0     540000.0   

3  Operating Profit  Operating Margin Sales Method  Year  Month  Day  
0          300000.0              0.50     In-store  2020      1    1  
1          150000.0              0.30     In-store  2020      1    2  
2          140000.0              0.35     In-store  2020      1    3  
3          133875.0              0.35     In-store  2020      1    4  
4          162000.0              0.30     In-store  2020      1    5

In [7]:
#exporting to excel file
df.to_excel('AdidasUSSales.xlsx')

# 2. Building the Dashboard on Streamlit

NOTE: 
The following is the source code to build the dashboard on Streamlit. Hence, the codes are tailored specifically to run on the Streamlit app. 
They will not run on Jupyter Notebook.

In [ ]:
import streamlit as st
import pandas as pd
import inspect
import textwrap
import plotly_express as px


def show_code(demo):
    """Showing the code of the demo."""
    show_code = st.sidebar.checkbox("Show code", True)
    if show_code:
        # Showing the code of the demo.
        st.markdown("## Code")
        sourcelines, _ = inspect.getsourcelines(demo)
        st.code(textwrap.dedent("".join(sourcelines[1:])))
@st.cache_data
def load_data(path :str):
    data=pd.read_excel(path)
    return data


st.set_page_config(
    page_title="Adidas Sales Dashboard",
    page_icon="bar_chart:",
    layout='wide')
st.title("Dashboard and Summary")
st.write(''' Welcome To Adidas US Sales Dashboard''')


df=load_data('./AdidasUSSales.xlsx')
df.dropna()
# Convert 'Invoice Date' to datetime
df['Invoice Date'] = pd.to_datetime(df['Invoice Date'])
df['Cumulative Total Sales'] = df.groupby('Retailer')['Total Sales'].cumsum()
df['Cumulative Total Sales Month Wise']=df.groupby('Month')['Total Sales'].cumsum()
df['Cumulative Sales State Wise']=df.groupby('State')['Total Sales'].cumsum()
df['Operating Profit State Wise']=df.groupby('State')['Operating Profit'].cumsum()
# Convert numeric columns to numeric types
numeric_columns = ['Price per Unit', 'Units Sold', 'Total Sales', 'Operating Profit', 'Operating Margin']
df[numeric_columns] = df[numeric_columns].apply(pd.to_numeric, errors='coerce')

with st.expander("Data Preview"):
    st.dataframe(df)
st.write('''
         USE THE FOLLOWING TAB FOR FILTERING RETAILER WISE
         ''')


# Select retailer for the bar race chart
selected_retailers = st.multiselect("Select Retailers", df["Retailer"].unique(), default=df["Retailer"].unique())

# Filter DataFrame based on selected retailers
filtered_df = df[df["Retailer"].isin(selected_retailers)].sort_values(by="Total Sales", ascending=False)

col1, col2 = st.columns(2,gap="medium")
# Sunburst Chart
with col1:
    st.subheader('Interactive Sunburst Chart')
    sunburst_chart = px.sunburst(filtered_df, path=['Retailer', 'Product'], values='Total Sales',
                                 title='Sunburst Chart')
    st.plotly_chart(sunburst_chart, use_container_width=True)

# Market Share Pie Chart
with col2:
    st.subheader("Market Share")
    # Group the data by retailer and sum the total sales for each retailer
    retailer_sales = filtered_df.groupby('Retailer')['Total Sales'].sum()
    # Calculate the total sales of all retailers
    total_sales = retailer_sales.sum()
    # Calculate the market share of each retailer by dividing their total sales by the total sales of all retailers
    market_share = retailer_sales / total_sales
    # Create a pie chart using plotly
    fig = px.pie(market_share, values=market_share, names=market_share.index, title='Market Share of Retailers')
    # Show the plot
    st.plotly_chart(fig)

st.subheader("Bar Race Chart - Total Sales by Retailer")

# Create a placeholder for the bar race chart
bar_race_chart = st.empty()
# Bar Race Chart
fig = px.bar(
    filtered_df,
    x=["Operating Profit"],
    y="Retailer",
    orientation="h",
    text="Total Sales",
    title=f"Total Sales Race for {selected_retailers} Monthly Numbers",
    animation_frame="Month",
    labels={"Total Sales": "Total Sales ($)"},
)
# Display the bar race chart
bar_race_chart.plotly_chart(fig,use_container_width=True)
# Simulate the race with a play button
if st.button("Start Race"):
    for i in range(1, len(filtered_df) + 1):
        # Update the chart dynamically
        updated_race = px.bar(
            filtered_df.iloc[:i],
            x="Total Sales",
            y="Retailer",
            orientation="h",
            text="Total Sales",
            title=f"Total Sales Race for {selected_retailers}",
            labels={"Total Sales": "Total Revenue($)"},
                  
        )
        bar_race_chart.plotly_chart(updated_race)    
        

yearly_sales = df.groupby(['Year','Month'])['Total Sales'].sum().reset_index()
yearly_profit = df.groupby(['Year','Month'])['Operating Profit'].sum().reset_index()
st.subheader('Monthly Total Sales and Profit Over Years')

# Create two columns for layout
col1, col2 = st.columns(2,gap ="medium")

# Line chart for monthly total sales over years
line_chart_sales = px.line(
    yearly_sales,
    x='Month',
    y='Total Sales',
    color='Year',
    title='Monthly Total Sales Over Years',
    markers=True,
    template="simple_white"
)

# Display the sales line chart using col1.plotly_chart
col1.plotly_chart(line_chart_sales,)

# Line chart for monthly total profit over years
line_chart_profit = px.line(
    yearly_profit,
    x='Month',
    y='Operating Profit',
    color='Year',
    title='Monthly Total Profit Over Years',
    markers=True,
    template="simple_white"
   
)

# Display the profit line chart using col2.plotly_chart
col2.plotly_chart(line_chart_profit)

# Advanced Scatter Plot Matrix
st.subheader('Advanced Scatter Plot Matrix')
scatter_matrix = px.scatter_matrix(df, dimensions=["Retailer","Product","Cumulative Sales State Wise", "State"], color='Operating Profit State Wise',
                                  title='Advanced Scatter Plot Matrix',width=1200, height=1000)
st.plotly_chart(scatter_matrix, use_container_width=True)



![1](1.JPG)
![2](2.JPG)
![3](3.JPG)

# 3. Data Analysis

## 3.1. Market Share of Retailers

* Amazon has the largest market share, with a 27.00% market capture, while Walmart is in second place, with 24.50%. Sports Direct is in third place, capturing 20.3% market share.
* Kohl’s, Foot Locker, and West Gear have much smaller market shares, ranging from 8.29% to 11.30%.


![4](4.jpg)


## 3.2. Operating profit of Retailers (2020-2021)

* West Gear is in first place with an operating profit of USD 85.67 million
* With an operating profit of USD 80.72 million Foot Locker is placed second of the 6 retailers
* Walmart has the least operating profit of USD 25.78 million


![5](5.jpg)

# 3.3. Retailer wise Sales across different products

* West gear has highest no. of units sold of 625k across all product category with highest sales in Men’s Street Footwear segment of USD 53 Million sales
* Amazon has least no. of units sold 198k across all product categories with highest sales in Men’s Street Footwear segment of USD 22 Million sales 


![6.1](6.1.jpg)
![6.2](6.2.jpg)

# 3.4. Total Sales vs. Total Profits (2020-2021)

The sales and operating profits have shown significant growth from 2020 to 2021.

* Total Sales of USD 717.82 million and total operating profit of 268.75 million dollars reported in 12 months starting January 2021
* Total Sales of USD 182.08 million and total operating profit of 63.37 million dollars reported in 12 months starting January 2020



![7](7.jpg)

# 3.5. Regional & State Variations in Sales & Profit

* With total sales of 64.22 million dollars and operating profit at USD 23.33 million, New York State is in the top place when comparing the state-wise sales and operating profit
* Florida and California are closely vying for second place
* Nebraska is the state with least selling products, with USD 5.92 million in sales and with USD 2.44 million
* Highest no. of sales was done in west region amounting to USD 270 Million with operating profit of USD 89 million with 686k no of units sold, followed by Northeast region with USD 186 Million sales
* Midwest region reported the least no. of sales USD 135 Million and operating profit of USD 52 million with 391k no. of units sold


![8.1](8.1.jpg)
![8.2](8.2.jpg)



# 3.6. Product Category Breakdown

* The top selling category is Men's Street Footwear with no. of units sold 593k, followed by Men's Athletic Footwear with no. of units sold 435k
* Men's Apparel holds last place in terms of units sold i.e. 306k from 2020 to 2021


![9](9.jpg)


# 3.7. Product Category Analysis

* Top Selling Product Category: Men's Street Footwear is the top-selling category, followed by Men's Athletic Footwear.

* Retailer Performance: West Gear leads in operating profit, followed closely by Foot Locker and Sports Direct.

* State and City Sales Analysis: New York State has the highest total sales. Whereas, Nebraska is the least-selling state.

* Monthly Sales and Profit Trends: There is a signifcant growth in the total sales by 294% and total profits grow by 324% from 2020 to 2021.


# 4. Key Takeaways & Implications

## 4.1. Overall Insights

Understanding the top-performing product categories, geographical areas, and retailers is essential for making strategic decisions and projecting demand. Since the demand for a specific product category is often steady, demand forecasting using the static technique can be employed to estimate future demand based on historical patterns. 

These techniques stand out for being straightforward and dependent only on historical data, ignoring outside variables that could affect variations in demand. The only information available for forecasting is historical data. When slight demand fluctuations, static approaches may be sufficient to capture regular patterns, obviating the need for more complicated models. Static techniques are helpful when prioritizing short-term planning. They provide quick insights into current demand estimates and don't require a thorough analysis. 

## 4.2. Conclusion

### Market Share
* Amazon dominates the market with a significant lead over its competitors, capturing nearly 27% of the market share.
* Walmart follows closely behind with 24.5%, while Sports Direct occupies a solid third place with 20.3%.
* Other retailers like Kohl's, Foot Locker, and West Gear have much smaller market shares.

### Profitability
* Despite having a smaller market share compared to Amazon and Walmart, West Gear stands out with the highest operating profit of USD 85.66 million.
* Notably, Walmart's operating profit is significantly lower at USD 25.78 million, despite its second-place position in market share.

### Regional Performance
* New York State leads in terms of state-wise sales and operating profit, with USD 64.22 million in sales and USD 23.33 million in profit.
* Florida and California closely follow in second place, while Nebraska holds the bottom position with the least sales and profit.
* The West region records the highest overall sales and operating profit, while the Midwest region reports the least.

### Growth & Trends
* Sales and operating profits have shown significant growth from 2020 to 2021, indicating a positive upward trend for the company.
* Men's Street Footwear and Men's Athletic Footwear are the top-selling categories, suggesting a strong demand for casual and athletic footwear among consumers.
* Men's Apparel appears to be the least popular category in terms of unit sales.
* West Gear leads in total units sold across all product categories, driven by strong performance in the Men's Street Footwear segment.
* Amazon, despite having the lowest unit sales overall, still manages to secure significant revenue in the Men's Street Footwear segment.

## 4.3. Managerial Implications

### Market Share vs. Profitability
* Amazon's dominant market share doesn't guarantee profitability. Look beyond market share and focus on operational efficiency and strategic product focus to maximize profits like West Gear (highest operating margin).
* Smaller retailers can compete by carving out niche markets and implementing targeted marketing strategies to attract loyal customers.

### Category Trends
* Capitalize on the high demand for casual and athletic footwear (top-selling categories) by expanding product offerings and promotions in these segments.
* Diversify product portfolio beyond low-performing categories like men's apparel to mitigate risk and cater to broader customer tastes.

### Specific Actions
* Benchmark against West Gear: Analyze their cost-control measures, efficient operations, and high-margin product focus to identify potential areas for improvement.
* Develop regional-specific marketing campaigns: Cater to local preferences and address unique needs of different demographics in each region.


# Acknowledgement

The code has been inspired from:
* Kaggle: Alameen Sajeev
* ChatGPT 